app/controllers/users_controller.rb

```diff
@@ -1,7 +1,57 @@
+# frozen_string_literal: true
+
 class UsersController < Clearance::UsersController
-  private
+  before_action :current_user, only: [:show, :edit, :update]
+  before_action :check_for_cancel, only: [:edit, :update]
+
+  # Could be useful for a future issue
+  # before_action :another_user
+
+  # def another_user
+  #   unless user_url.end_with?(current_user.id.to_s)
+  #     redirect_to request.referer
+  #   end
+  # end
+
+  def check_for_cancel
+    if params[:commit] == "Cancel"
+      redirect_to user_url
+    end
+  end
+
+  def index
+    @users = User.all
+  end
+
+  def show
```

Comment:
I'd suggest to use `index` instead of `show` for showing user's data.

There is no need for exposing or requiring the id in the URL 😄

Reply:
Oh, we had problems showing the user's data because it's possible to see another user's data from another account, so if we use index, is it possible to fix that problem? 😅

Reply:
We will implement friendlyId gem in another issue to change the URLs

Reply:
You don't need the `friendlyId` gem at all. Neither the user in the URL, since you are always able to retrieve the `user` via authentication. Every time you call `current_user` in a controller that requires authentication, you are querying for the logged-user. Ideally, user's should be only able to edit it's own information, right?

Am I missing something?

Reply:
We don't know how to change the URL or remove the id from it, that's why we want to use that gem.

Reply:
I see.

Let's pair on this.
Sadly, I think I can't do it today, and I can help you to get this done :smile:

Just create an issue for the follow-up. Otherwise, this refactor would be forgot 😂

Reply:
Sure, thanks!